public/app/features/dashboard/state/PanelQueryRunner.ts

```diff
@@ -118,19 +111,19 @@ export class PanelQueryRunner {
       maxDataPoints,
       scopedVars,
       minInterval,
-      delayStateNotification,
+      // delayStateNotification,
     } = options;
 
     // Support shared queries
-    if (isSharedDashboardQuery(datasource)) {
-      if (!this.sharedQueryRunner) {
-        this.sharedQueryRunner = new SharedQueryRunner(this);
-      }
-      return this.sharedQueryRunner.process(options);
-    } else if (this.sharedQueryRunner) {
-      this.sharedQueryRunner.disconnect();
-      this.sharedQueryRunner = null;
-    }
+    // if (isSharedDashboardQuery(datasource)) {
```

Comment:
For shared queries, is your plan to subscribe to the original panel's runner instead of creating a custom one?

Reply:
I have a new implementation for shared queries (a simple function that returns Observable<PanelData>, pretty happy 
https://github.com/grafana/grafana/blob/query-observable/public/app/plugins/datasource/dashboard/runSharedRequest.ts 

this part in PanelQueryRuner is now just
<img width="644" alt="Screen Shot 2019-09-06 at 14 01 31" src="https://user-images.githubusercontent.com/10999/64426472-de102e00-d0ae-11e9-93c5-f1d466282d67.png">